hudi-flink/src/main/java/org/apache/hudi/util/AvroSchemaConverter.java

```diff
@@ -244,10 +244,13 @@ public static Schema convertToSchema(LogicalType logicalType, String rowName) {
         return nullable ? nullableSchema(time) : time;
       case DECIMAL:
         DecimalType decimalType = (DecimalType) logicalType;
-        // store BigDecimal as byte[]
+        // store BigDecimal as Fixed
+        // for spark compatibility.
         Schema decimal =
             LogicalTypes.decimal(decimalType.getPrecision(), decimalType.getScale())
-                .addToSchema(SchemaBuilder.builder().bytesType());
+                .addToSchema(SchemaBuilder
+                    .fixed(String.format("%s.fixed", rowName))
+                    .size(computeMinBytesForDecimlPrecision(decimalType.getPrecision())));
```

Comment:
Thanks for the fix, can this type be read through Hive directly ? There is also some type compatibility for Spark & Hive decimal types.

Reply:
Yes, hive can be read directly. Hive has good compatibility. Forgive me, I can't take a screenshot here because of the company's regulations
Spark saves decimal as int / long / byte [] / fixed according to precision, of which the fourth is the most compatible。

Reply:
Some of the community users feedback that the hive 2 and 3 has different physical type for decimal, the low version hive seems to have `bytes` as parquet physical type and the high version hive has `fixed_bytes` as the physical type.

Reply:
Thank you for your reminder. let me find a hive2 environment to verify this pr.
by the way， how about control this behavior by configurations item